dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/business/HostAPITest.java

```diff
@@ -639,4 +639,25 @@ private void addPermission(final Role role, final Host host)
         APILocator.getPermissionAPI().save(CollectionsUtils.list(permission), host, systemUser, false);
     }
 
+    /**
+     * Method to test: {@link HostAPI#findByAlias(String, User, boolean)}
+     * When create two host: first one with alias equals to demo.dotcms.com and second  one with alias equals to not-demo.dotcms.com
+     *      and find by  demo.dotcms.com
+     * Should return the first one
+     */
+    @Test
+    public void shouldReturnHostByAlias() throws DotSecurityException, DotDataException {
+        final Host host = new SiteDataGen().aliases("demo.dotcms.com").nextPersisted();
+        final Host host_2 = new SiteDataGen().aliases("not-demo.dotcms.com").setDefault(true).nextPersisted();
```

Comment:
Please write a test that tests where there are multiple host aliases separated by \r and \n and also a tab and space. 

Also, the broken case is where there are multiple hosts that start with `not-` e.g. `prod-client.dotcms.com` and `prod-anotherclient.dotcms.com`.  Please write a test for this as well - as it is the broken case.

Reply:
done

https://github.com/dotCMS/core/pull/18829/commits/1c06c59de80cc2b15782eceb455d7b8d27d149ec